src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionAnnotatedTypeFactory.java

```diff
@@ -1,18 +1,505 @@
+            "to",
+            processingEnv,
+            "com.google.inject.Key<? extends T>"));
+
+    this.toInstanceMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.LinkedBindingBuilder", "toInstance", processingEnv, "T"));
+
+    this.annotatedWithMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.AnnotatedBindingBuilder",
+            "annotatedWith",
+            processingEnv,
+            "java.lang.annotation.Annotation"));
+    this.annotatedWithMethods.add(
+        TreeUtils.getMethod(
+            "com.google.inject.binder.AnnotatedBindingBuilder",
+            "annotatedWith",
+            processingEnv,
+            "java.lang.Class<? extends java.lang.annotation.Annotation>"));
+
+    this.postInit();
+  }
+
+  /* Returns true iff the argument is an invocation of AbstractModule.bind.
+   *
+   * @param methodTree the method invocation tree
+   * @return true iff the argument is an invocation of AbstractModule.bind()
+   */
+  protected boolean isBindMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(methodTree, this.bindMethods, this.getProcessingEnv());
+  }
+
+  /* Returns true iff the argument is an invocation of LinkedBindingBuilder.to() */
+  protected boolean isToMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(methodTree, this.toMethods, this.getProcessingEnv());
+  }
+
+  /* Returns true iff the argument is an invocation of LinkedBindingBuilder.toInstance() */
+  protected boolean isToInstanceMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(
+        methodTree, this.toInstanceMethods, this.getProcessingEnv());
+  }
+
+  /* Returns true iff the argument is an invocation of AnnotatedBindingBuilder.annotatedWith() */
+  protected boolean isAnnotatedWithMethod(Tree methodTree) {
+    return TreeUtils.isMethodInvocation(
+        methodTree, this.annotatedWithMethods, this.getProcessingEnv());
+  }
+
+  /* Invoked when the `MethodInvocationNode` is a call to `com.google.inject.AbstractModule.bind`
+
+  * This method will add the bound class to the `knownBindings` map
+  *
+  * @param methodArgumentNode the argument to the `bind` method
+  */
+  private void handleBindMethodInvocation(Node methodArgumentNode) {
+    // Class that is being bound - put in knownBindings
+    AnnotatedTypeMirror boundClassTypeMirror =
+        classValATF.getAnnotatedType(methodArgumentNode.getTree());
+
+    // TODO: getAnnotation may possibly return annotations that aren't @ClassVal
+    List<String> classNames =
+        AnnotationUtils.getElementValueArray(
+            boundClassTypeMirror.getAnnotation(), classValValueElement, String.class);
+
+    classNames.forEach(
+        className -> {
+          DependencyInjectionAnnotatedTypeFactory.knownBindings.put(className, null);
+        });
+  }
+
+  /* Invoked when the `MethodInvocationNode` is a call to `com.google.inject.binder.LinkedBindingBuilder.to`
```

Comment:
Just went through this and remembered that the node that is passed to this method is not actually a `MethodInvocationNode`, but it's argument. I mention this because the place in which it is called has to check whether the expression is a `MethodInvocationNode` first and if it is it extracts its argument and calls this method.

Reply:
I think the documentation should be updated to make this clear, then: right now, it's a bit confusing that a `MethodInvocationNode` is mentioned, but that the parameter isn't one. In particular, the Javadoc for the parameter should have this sentence from your answer to me:
```
the node that is passed to this method is not actually a MethodInvocationNode, but its argument
```